crates/sui-graphql-rpc/src/types/transaction_block_kind/programmable.rs

```diff
@@ -0,0 +1,486 @@
+}
+
+/// Splits off coins with denominations in `amounts` from `coin`, returning multiple results (as
+/// many as there are amounts.)
+#[derive(SimpleObject, Clone, Eq, PartialEq)]
+struct SplitCoinsTransaction {
+    /// The coin to split.
+    coin: TransactionArgument,
+
+    /// The denominations to split off from the coin.
+    amounts: Vec<TransactionArgument>,
+}
+
+/// Merges `coins` into the first `coin` (produces no results).
+#[derive(SimpleObject, Clone, Eq, PartialEq)]
+struct MergeCoinsTransaction {
+    /// The coin to merge into.
+    coin: TransactionArgument,
+
+    /// The coins to be merged.
+    coins: Vec<TransactionArgument>,
+}
+
+/// Publishes a Move Package.
+#[derive(SimpleObject, Clone, Eq, PartialEq)]
+struct PublishTransaction {
+    /// Bytecode for the modules to be published, BCS serialized and Base64 encoded.
+    modules: Vec<Base64>,
+
+    /// IDs of the transitive dependencies of the package to be published.
+    dependencies: Vec<SuiAddress>,
+}
+
+/// Upgrades a Move Package.
+#[derive(SimpleObject, Clone, Eq, PartialEq)]
+struct UpgradeTransaction {
+    /// Bytecode for the modules to be published, BCS serialized and Base64 encoded.
+    modules: Vec<Base64>,
+
+    /// IDs of the transitive dependencies of the package to be published.
+    dependencies: Vec<SuiAddress>,
+
+    /// ID of the package being upgraded.
+    current_package: SuiAddress,
+
+    /// The `UpgradeTicket` authorizing the upgrade.
+    upgrade_ticket: TransactionArgument,
+}
+
+/// Create a vector (possibly empty).
+#[derive(SimpleObject, Clone, Eq, PartialEq)]
+struct MakeMoveVecTransaction {
+    /// If the elements are not objects, or the vector is empty, a type must be supplied.
+    #[graphql(name = "type")]
+    type_: Option<MoveType>,
+
+    /// The values to pack into the vector, all of the same type.
+    elements: Vec<TransactionArgument>,
+}
+
+/// An argument to a programmable transaction command.
+#[derive(Union, Clone, Eq, PartialEq)]
+enum TransactionArgument {
+    GasCoin(GasCoin),
+    Input(Input),
+    Result(TxResult),
+}
+
+/// Access to the gas inputs, after they have been smashed into one coin. The gas coin can only be
+/// used by reference, except for with `TransferObjectsTransaction` that can accept it by value.
+#[derive(SimpleObject, Clone, Eq, PartialEq)]
+struct GasCoin {
+    /// A workaround to define an empty variant of a GraphQL union.
+    #[graphql(name = "_")]
+    dummy: Option<bool>,
+}
+
+/// One of the input objects or primitive values to the programmable transaction block.
+#[derive(SimpleObject, Clone, Eq, PartialEq)]
+struct Input {
+    /// Index of the programmable transaction block input (0-indexed).
+    ix: u16,
+}
+
+/// The result of another transaction command.
+#[derive(SimpleObject, Clone, Eq, PartialEq)]
+#[graphql(name = "Result")]
+struct TxResult {
+    /// The index of the previous command (0-indexed) that returned this result.
+    cmd: u16,
```

Comment:
nit, feel free to disregard.
```suggestion
    cmd_ix: u16,
```